ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2

```diff
@@ -80,6 +80,33 @@ interface {{ port_channel_interface }}
 {%         if port_channel_interfaces[port_channel_interface].rt is arista.avd.defined %}
       route-target import {{ port_channel_interfaces[port_channel_interface].rt }}
 {%         endif %}
+{%         if port_channel_interfaces[port_channel_interface].evpn_ethernet_segment.redundancy is arista.avd.defined %}
+      redundancy {{ port_channel_interfaces[port_channel_interface].evpn_ethernet_segment.redundancy }}
+{%         endif %}
+{%         if port_channel_interfaces[port_channel_interface].evpn_ethernet_segment.designated_forwarder_election is arista.avd.defined %}
+{%             set designated_forwarder_cli = "designated-forwarder election" %}
+{%             if port_channel_interfaces[port_channel_interface].evpn_ethernet_segment.designated_forwarder_election.algorithm is arista.avd.defined("preference") %}
+{%                set designated_forwarder_algorithm_cli = designated_forwarder_cli ~ " algorithm " ~ port_channel_interfaces[port_channel_interface].evpn_ethernet_segment.designated_forwarder_election.algorithm ~ " " ~ port_channel_interfaces[port_channel_interface].evpn_ethernet_segment.designated_forwarder_election.preference_value %}
```

Comment:
Sure, will add var check as follows:

```yaml
{%             if port_channel_interfaces[port_channel_interface].evpn_ethernet_segment.designated_forwarder_election.algorithm is arista.avd.defined("preference") and port_channel_interfaces[port_channel_interface].evpn_ethernet_segment.designated_forwarder_election.preference_value is arista.avd.defined %}

```

Reply:
Also, you checked ```port_channel_interfaces[port_channel_interface].evpn_ethernet_segment.designated_forwarder_election.algorithm``` is ```preference``` so directly use the value in line 89 instead of using the variable.